src/Build/BackEnd/Components/Scheduler/Scheduler.cs

```diff
@@ -1781,6 +1786,74 @@ private ScheduleResponse TrySatisfyRequestFromCache(int nodeForResults, BuildReq
             return null;
         }
 
+        private bool EnforceCacheMissInIsolatedProjectBuilds(int nodeForResults, BuildRequest request, List<ScheduleResponse> responses)
```

Comment:
The semantics of this confused me because I interpreted it backward. Maybe rename to `CacheMissIsAllowed()`?